### PR TITLE
Implement docs generation pipeline on Buildkite

### PR DESCRIPTION
**Pull request by fweikert**
After this change we can replace our CloudBuild pipelines with Buildkite pipelines. They'll run as GitHub postsubmit and create and upload new versions of the Bazel documentation, the blog and the website.
Moreover, this code will be used by a "docs staging" presubmit pipeline, which still needs some work, though.

Next steps:
1. Create Buildkite pipelines
2. Disable CloudBuild pipelines
3. Enable Buildkite pipelines
4. Update (internal) documentation
5. Remove obsolete CloudBuild configs 
6. Fix and enable staging pipeline

Progress towards #1110 (Milestone 1)

### CODE REVIEW

File: docgen/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+ENV DEBIAN_FRONTEND="noninteractive"
+RUN apt-get -qqy update && \
+    apt-get -qqy install build-essential curl liblzma-dev \
+      python3.7 python-pygments ruby ruby-dev unzip zlib1g-dev && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+RUN ln -fs /usr/bin/python3.7 /usr/bin/python
+
+### Install packages required by bazelci.pys
```
Comment (philwo) [inline docgen/Dockerfile:16]: What's `pys`?

File: docgen/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+### https://cloud.google.com/sdk/docs/quickstart-debian-ubuntu
+RUN export CLOUD_SDK_REPO="cloud-sdk-bionic" && \
+    echo "deb http://packages.cloud.google.com/apt $CLOUD_SDK_REPO main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list && \
+    curl -L https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add - && \
+    apt-get -qqy update && \
+    apt-get -qqy install google-cloud-sdk && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY Gemfile .
+RUN gem install -g --no-rdoc --no-ri && rm -f Gemfile
```
Comment (philwo) [inline docgen/Dockerfile:42]: Could you (also in other files) add a final newline in text files?

Comment (philwo) [reply docgen/Dockerfile:42]: If you're using VS Code, this is my favorite config to ensure clean files:
Screenshot 2021-02-22 at 15 26 21

**Review by philwo — APPROVED** [review]

File: docgen/Dockerfile (hunk 1 above)
Comment (fweikert) [reply docgen/Dockerfile:16]: typo

**Pull request merged**